satpy/tests/reader_tests/test_seviri_l1b_native.py

```diff
@@ -548,177 +553,83 @@ def test_earthmodel2_hrv_roi(self):
-                        'GSICSOffsetCount': [-51.0, -51.0, -51.0, -51.0, -51.0, -51.0,
-                                             -51.0, -51.0, -51.0, -51.0, -51.0, -51.0]
-                    },
+                        'GSICSCalCoeff': self.gains_gsics,
+                        'GSICSOffsetCount': self.offsets_gsics
+                    }
                 },
-                'SatelliteStatus': {
-                    'SatelliteDefinition': {
-                        'SatelliteId': 324
+                'ImageDescription': {
+                    'Level15ImageProduction': {
+                        'PlannedChanProcessing': self.radiance_types
+                    }
+                },
+                'ImageAcquisition': {
+                    'PlannedAcquisitionTime': {
+                        'TrueRepeatCycleStart': self.scan_time
                     }
                 }
-            },
-            '15_SECONDARY_PRODUCT_HEADER': {
-                'NorthLineSelectedRectangle': {'Value': north},
-                'EastColumnSelectedRectangle': {'Value': east},
-                'WestColumnSelectedRectangle': {'Value': west},
-                'SouthLineSelectedRectangle': {'Value': south},
-                'SelectedBandIDs': {'Value': 'xxxxxxxxxxxx'},
-                'NumberColumnsVISIR': {'Value': n_visir_cols},
-                'NumberLinesVISIR': {'Value': n_visir_lines},
-                'NumberColumnsHRV': {'Value': 11136},
-                'NumberLinesHRV': {'Value': n_hrv_lines},
             }
-
         }
+        with mock.patch('satpy.readers.seviri_l1b_native.NativeMSGFileHandler.__init__',
+                        return_value=None):
+            fh = NativeMSGFileHandler()
+            fh.header = header
+            fh.platform_id = self.platform_id
+            return fh
+
+    @pytest.mark.parametrize(
+        ('channel', 'calibration', 'calib_mode', 'use_ext_coefs'),
+        [
+            # VIS channel, internal coefficients
+            ('VIS006', 'counts', 'NOMINAL', False),
+            ('VIS006', 'radiance', 'NOMINAL', False),
+            ('VIS006', 'radiance', 'GSICS', False),
+            ('VIS006', 'reflectance', 'NOMINAL', False),
+            # VIS channel, external coefficients (mode should have no effect)
+            ('VIS006', 'radiance', 'GSICS', True),
+            ('VIS006', 'reflectance', 'NOMINAL', True),
+            # IR channel, internal coefficients
+            ('IR_108', 'counts', 'NOMINAL', False),
+            ('IR_108', 'radiance', 'NOMINAL', False),
+            ('IR_108', 'radiance', 'GSICS', False),
+            ('IR_108', 'brightness_temperature', 'NOMINAL', False),
+            ('IR_108', 'brightness_temperature', 'GSICS', False),
+            # IR channel, external coefficients (mode should have no effect)
+            ('IR_108', 'radiance', 'NOMINAL', True),
+            ('IR_108', 'brightness_temperature', 'GSICS', True),
+        ]
```

Comment:
The parametrization is the same for HRIT and Native, but differs for netCDF (which has only one calibration mode). Should it be factorized anyway?